File: languages/ruby/tak.rb
$calls = 0

def tak(x, y, z)
  $calls += 1
  return tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y)) if y < x
  z
end

n = ARGV[0] ? ARGV[0].to_i : 6
r = tak(3 * n, 2 * n, n)
puts $calls
puts "tak(#{n}) = #{r}"

File: languages/perl/tak.pl
use strict;
use warnings;

my $calls = 0;

sub tak {
    my ($x, $y, $z) = @_;
    $calls++;
    return $z if $y >= $x;
    return tak(tak($x - 1, $y, $z), tak($y - 1, $z, $x), tak($z - 1, $x, $y));
}

my $n = @ARGV ? int($ARGV[0]) : 6;
my $r = tak(3 * $n, 2 * $n, $n);
print "$calls\n";
print "tak($n) = $r\n";
